### ISSUE TITLE
New quests not marked as completed

### ISSUE THREAD

**Issue by sinsinpub** [KC3Kai/KC3Kai#1528]
Recently added new quests will not be marked as completed by KC3 when quest indeed completed in game, because `api_state` of `kcsapi/api_get_member/questlist` returns value `2` instead of `3`.

For example, my quest id 644 (「一式陸攻」性能向上型の調達) is completed, but API data:
```json
svdata={"api_result":1,"api_result_msg":"\u6210\u529F","api_data":{"api_count":17,"api_completed_kind":1,"api_page_count":4,"api_disp_page":4,"api_list":[{"api_no":639,"api_category":6,"api_type":4,"api_state":1,"api_title":"\u65B0\u578B\u9B5A\u96F7\u5175\u88C5\u306E\u958B\u767A","api_detail":"\u52F2\u7AE0x2\u6D88\u8CBB\uFF1A\u300C\u5CF6\u98A8\u300D\u79D8\u66F8\u8266\u306B\u6539\u4FEEmax\u300C61cm\u4E94\u9023\u88C5(\u9178\u7D20)\u9B5A\u96F7\u300D\u3068<br>\u6539\u4FEEmax\u300C61cm\u4E09\u9023\u88C5(\u9178\u7D20)\u9B5A\u96F7\u300D\u3092\u88C5\u5099\uFF01(\u4EFB\u52D9\u9054\u6210\u5F8C\u3001\u5404\u88C5\u5099\u306F\u6D88\u6EC5\u3057\u307E\u3059)","api_lost_badges":2,"api_get_material":[0,600,0,0],"api_bonus_flag":1,"api_progress_flag":0,"api_invalid_flag":0},{"api_no":644,"api_category":6,"api_type":4,"api_state":2,"api_title":"\u300C\u4E00\u5F0F\u9678\u653B\u300D\u6027\u80FD\u5411\u4E0A\u578B\u306E\u8ABF\u9054","api_detail":"\u300C\u4E00\u5F0F\u9678\u653B\u300D\u6027\u80FD\u5411\u4E0A\u578B\u306E\u914D\u5099\u3092\u884C\u3046\u3002\u300C\u4E00\u5F0F\u9678\u653B\u300D\u4E00\u3064\u3068\u300C\u5929\u5C71\u300D\u4E8C\u3064\u3092\u6E96\u5099\u305B\u3088\uFF01<br>(\u4EFB\u52D9\u9054\u6210\u5F8C\u3001\u7528\u610F\u3057\u305F\u5FC5\u8981\u88C5\u5099\u306F\u6D88\u6EC5\u3057\u307E\u3059/\u300C\u4E00\u5F0F\u9678\u653B\u300D\u306E\u719F\u7DF4\u5EA6\u306F\u7D99\u627F\u3055\u308C\u307E\u3059)","api_get_material":[0,0,0,300],"api_bonus_flag":1,"api_progress_flag":0,"api_invalid_flag":0},-1,-1,-1],"api_exec_count":6,"api_exec_type":2499816}}
```

another solution may needs to be found for this :sweat:

**Comment by sinsinpub** [comment]
seems #1521 caused by the same reason

**Comment by dragonjet** [comment]
maybe it's the presence of `api_exec_count` and the other exec field?

**Comment by sinsinpub** [comment]
I've compared with other old quests (both opened and completed), they are no difference...
I guess the 'completed state' is judged by client-side (actionscript) based on the equipment player holding, so not included in server-side API response.

**Comment by Javran** [comment]
this might be useful: response upon completing F41:

> http://<server>/kcsapi/api_req_quest/clearitemget
> api_verno:1
> api_quest_id:645

response:
```json
{
	"api_result": 1,
	"api_result_msg": "成功",
	"api_data": {
		"api_material": [
			0,
			0,
			0,
			0
		],
		"api_bounus_count": 1,
		"api_bounus": [
			{
				"api_type": 15,
				"api_count": 1,
				"api_item": {
					"api_id_from": 9999,
					"api_id_to": 146,
					"api_message": "「洋上補給」物資を調達完了！<br>(※使用には補給艦による運用が必要です)"
				}
			}
		]
	}
}
```

**Comment by dragonjet** [comment]
fix on `bd74a13` is untested. will have to re-open this if found still persists
